File: PetitionBackGroundService/kafkaConfig.js
require("dotenv").config();
const { Kafka } = require("kafkajs");

const kafka = new Kafka({
  clientId: "my-app",
  brokers: [process.env.KAFKA_BROKER],
  ssl: {},
  sasl: {
    mechanism: "scram-sha-256",
    username: process.env.KAFKA_USERNAME,
    password: process.env.KAFKA_PASSWORD,
  },
});

module.exports = kafka;

File: Categorization/kafka_config.py
from confluent_kafka import Producer
import os
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

KAFKA_CONFIG = {
    "bootstrap.servers": os.getenv("KAFKA_BROKER"),  # e.g., "your-kafka-broker:9092"
    "security.protocol": "SASL_SSL",
    "sasl.mechanism": "SCRAM-SHA-256",
    "sasl.username": os.getenv("KAFKA_USERNAME"),
    "sasl.password": os.getenv("KAFKA_PASSWORD"),
}

def get_producer():
    return Producer(KAFKA_CONFIG)
